wheel/wheelfile.py

```diff
@@ -63,9 +63,11 @@ def __init__(self, file, mode='r'):
                     path, hash_sum, size = line.rsplit(u',', 2)
                     if hash_sum:
                         algorithm, hash_sum = hash_sum.split(u'=')
-                        if algorithm not in hashlib.algorithms_available:
+                        try:
+                            hashlib.new(algorithm)
+                        except ValueError:
                             raise WheelError('Unsupported hash algorithm: {}'.format(algorithm))
-                        elif algorithm.lower() in {'md5', 'sha1'}:
+                        if algorithm.lower() in {'md5', 'sha1'}:
```

Comment:
Small nit: add an empty line after the try...except block. After that I'll merge.

Reply:
Done.